src/main/java/core/org/akaza/openclinica/web/filter/OpenClinicaOAuthRequestAuthenticator.java

```diff
@@ -0,0 +1,57 @@
+package core.org.akaza.openclinica.web.filter;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.adapters.KeycloakDeployment;
+import org.keycloak.adapters.OAuthRequestAuthenticator;
+import org.keycloak.adapters.OIDCAuthenticationError;
+import org.keycloak.adapters.RequestAuthenticator;
+import org.keycloak.adapters.spi.AdapterSessionStore;
+import org.keycloak.adapters.spi.AuthChallenge;
+import org.keycloak.adapters.spi.HttpFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+public class OpenClinicaOAuthRequestAuthenticator extends OAuthRequestAuthenticator {
```

Comment:
Add javadoc on the class explaining why we are customizing the request authenticator.